src/main/java/org/myrobotlab/service/InMoov2.java

```diff
@@ -596,82 +486,55 @@ public String get(String key) {
   }
 
   public InMoov2Arm getArm(String side) {
-    if ("left".equals(side)) {
-      return leftArm;
-    } else if ("right".equals(side)) {
-      return rightArm;
-    } else {
-      log.error("can not get arm {}", side);
-    }
-    return null;
+    return (InMoov2Arm) getPeer(side + "Arm");
   }
 
   public InMoov2Hand getHand(String side) {
-    if ("left".equals(side)) {
-      return leftHand;
-    } else if ("right".equals(side)) {
-      return rightHand;
-    } else {
-      log.error("can not get arm {}", side);
-    }
-    return null;
+    return (InMoov2Hand) getPeer(side + "Hand");
   }
 
   public InMoov2Head getHead() {
-    return head;
+    return (InMoov2Head) getPeer("head");
   }
 
   /**
    * finds most recent activity
    * 
    * @return the timestamp of the last activity time.
    */
-  public long getLastActivityTime() {
-
-    long lastActivityTime = 0;
-
-    if (leftHand != null) {
-      lastActivityTime = Math.max(lastActivityTime, leftHand.getLastActivityTime());
-    }
-
-    if (leftArm != null) {
-      lastActivityTime = Math.max(lastActivityTime, leftArm.getLastActivityTime());
-    }
-
-    if (rightHand != null) {
-      lastActivityTime = Math.max(lastActivityTime, rightHand.getLastActivityTime());
-    }
+  public Long getLastActivityTime() {
+    try {
 
-    if (rightArm != null) {
-      lastActivityTime = Math.max(lastActivityTime, rightArm.getLastActivityTime());
-    }
+      Long lastActivityTime = 0L;
 
-    if (head != null) {
-      lastActivityTime = Math.max(lastActivityTime, head.getLastActivityTime());
-    }
+      Long head = (Long) sendToPeerBlocking("head", "getScript", getName());
+      Long leftArm = (Long) sendToPeerBlocking("leftArm", "getScript", getName());
+      Long rightArm = (Long) sendToPeerBlocking("rightArm", "getScript", getName());
+      Long leftHand = (Long) sendToPeerBlocking("leftHand", "getScript", getName());
+      Long rightHand = (Long) sendToPeerBlocking("rightHand", "getScript", getName());
+      Long torso = (Long) sendToPeerBlocking("torso", "getScript", getName());
 
-    if (torso != null) {
-      lastActivityTime = Math.max(lastActivityTime, torso.getLastActivityTime());
-    }
+      lastActivityTime = Math.max(head, leftArm);
+      lastActivityTime = Math.max(lastActivityTime, rightArm);
+      lastActivityTime = Math.max(lastActivityTime, leftHand);
+      lastActivityTime = Math.max(lastActivityTime, rightHand);
+      lastActivityTime = Math.max(lastActivityTime, torso);
 
-    if (lastPirActivityTime != null) {
-      lastActivityTime = Math.max(lastActivityTime, lastPirActivityTime);
-    }
+      return lastActivityTime;
 
-    if (lastActivityTime == 0) {
-      error("invalid activity time - anything connected?");
-      lastActivityTime = System.currentTimeMillis();
+    } catch (Exception e) {
+      error(e);
```

Comment:
add an actual error message if we actually need this exception handling.  what would throw this?

Reply:
send blocking potentially can timeout

Reply:
error(e) should be enough - it should give a message and a stack trace